zenoh/src/lib.rs

```diff
@@ -109,8 +109,24 @@ pub const FEATURES: &str = zenoh_util::concat_enabled_features!(
     ]
 );
 
-// Expose some functions directly to root `zenoh::`` namespace for convenience
-pub use crate::api::{scouting::scout, session::open};
+#[doc(inline)]
+pub use crate::{
```

Comment:
I'm not against but not 100% convinced wether it's a good idea to export all those types directly under `zenoh::*`. @milyin any opinion?

Reply:
Being able to write `zenoh::Result`/`zenoh::Error` is kind of mandatory IMO. 
Also a lot of common english names, you may not want to import them directly, but use the module instead, for example `zenoh::Session`, or `zenoh::Priority`. I think it's more convenient for the user (in my own code, I think I would use `zenoh::Session` more than `Session`).

Reply:
I agree that some user may prefer `zenoh::Session` to just `Session`. My point is more on what do we decide to put in the root? And what not? E.g., `Session`, `Result`, `Error` make sense to be in the root but I'm not sure about `Priority` (which is just taken as an example here). Is `Priority` really a first class type to be exported in the root? Are we going to put all types in the root loosing any kind of hierarchy/grouping?

Reply:
I can split the PR in two, one for the prelude, and one draft for the putting the types into the root. The only issue is about imports in the examples.